js/test/utils.test.js

```diff
@@ -1,24 +1,39 @@
 /* global BigInt describe it */
 
 import * as chai from 'chai';
-import * as utils from '../utils/utils';
+import * as utils from '../src/utils';
+import * as vectors from '../../testVectors.json';
+// import { type } from 'os';
+
+const vectorObj = JSON.parse(JSON.stringify(vectors));
+utils.parseJson(vectorObj);
+
+const {
+  lastBytes,
+  reverseEndianness,
+  bytesToUint,
+  sha256,
+  ripemd160,
+  typedArraysAreEqual,
+  safeSlice
+} = vectorObj;
 
 const { assert } = chai;
 
 describe('utils', () => {
   describe('#lastBytes', () => {
     it('gets the last bytes correctly', () => {
-      const res = utils.lastBytes(utils.deserializeHex('0x00112233'), 2);
-      const arraysAreEqual = utils.typedArraysAreEqual(res, utils.deserializeHex('0x2233'));
+      const res = utils.lastBytes(lastBytes[0].input.proof, lastBytes[0].input.num);
```

Comment:
we can roll several of these tests into a for loop instead. same goes for several other places